app/Models/Bible/BibleBook.php

```diff
@@ -119,4 +138,51 @@ public function book()
     {
         return $this->belongsTo(Book::class);
     }
+
+    /**
+     * Get collection of bible books sorted by bible versification or the book_seq column if it not empty.
+     *
+     * @param string $bible_id
+     * @param string $book_id
+     * @param string $bible_versification
+     *
+     * @return Collection
+     */
+    public static function getAllSortedByBookSeqOrVersification(
+        string $bible_id,
+        ?string $book_id = null,
+        ?string $bible_versification = null
```

Comment:
bible_versification will always be non-null

Reply:
@bradflood , I have updated it and for now on, the bible_versification parameter must be passed.